### PR TITLE
Allow strings for boolean workflow step parameters

### LINKED ISSUE #669: [BUG] Handling of Booleans should be robust to strings
### What is the bug?

Using `"true"` in some workflow steps when a boolean `true` is expected causes a 500 error (ClassCastException).

### How can one reproduce the bug?

Add a template to register a local model with `"deploy": "true"`. 

### What is the expected behavior?

The template parsing interprets the String `"true"` as Boolean.TRUE.

### Do you have any additional context?

These four instance of casting `(Boolean) inputs.get(...)` should be changed to use `Boolean.parseBoolean(inputs.get(...).toString())`:

https://github.com/search?q=repo%3Aopensearch-project%2Fflow-framework%20%22(Boolean)%20inputs.get%22&type=code

Alternately consider opensearch common `Booleans` class which throws an error for anything other than "true" or "false".

Unit tests should be added for both types of params.

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description
Allows `"true"` or `"false"` strings when boolean parameters are expected.

### Issues Resolved

Fixes #669 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/flowframework/workflow/RegisterRemoteModelStep.java [hunk 1]
```diff
@@ -90,7 +92,7 @@ public PlainActionFuture<WorkflowData> execute(
             String description = (String) inputs.get(DESCRIPTION_FIELD);
             String connectorId = (String) inputs.get(CONNECTOR_ID);
             Guardrails guardRails = (Guardrails) inputs.get(GUARDRAILS_FIELD);
-            final Boolean deploy = (Boolean) inputs.get(DEPLOY_FIELD);
+            final Boolean deploy = inputs.containsKey(DEPLOY_FIELD) ? Booleans.parseBoolean(inputs.get(DEPLOY_FIELD).toString()) : null;
```
Comment (owaiskazi19) [inline src/main/java/org/opensearch/flowframework/workflow/RegisterRemoteModelStep.java:95]: Why a special check just for the boolean type?

Shouldn't it be false?
```suggestion
            final Boolean deploy = inputs.containsKey(DEPLOY_FIELD) ? Booleans.parseBoolean(inputs.get(DEPLOY_FIELD).toString()) : false;
```

or simply
```suggestion
            final Boolean deploy = Booleans.parseBoolean(inputs.get(DEPLOY_FIELD).toString());
```

Comment (dbwiddis) [reply src/main/java/org/opensearch/flowframework/workflow/RegisterRemoteModelStep.java:95]: > Shouldn't it be false?

Existing behavior is to cast the result of `Map.get()`.  If the value isn't present, the result is `(Boolean) null` not `Boolean.FALSE`.  In theory we could hardcode the "default" but then we'd have to make sure that whichever way we are paring the boolean, we match the default.  False here, true for some of the other fixes in this PR.

Comment (dbwiddis) [reply src/main/java/org/opensearch/flowframework/workflow/RegisterRemoteModelStep.java:95]: `Booleans.parseBoolean(inputs.get(DEPLOY_FIELD).toString())` will fail with an NPE if the key is not present.  We'd need to use `getOrDefault()`.  I did consider that, but that does require us to know what the default is (see above comment) and from a DRY principle I'd prefer to keep the result the same.... null.

File: src/test/java/org/opensearch/flowframework/workflow/RegisterModelGroupStepTests.java [hunk 2]
```diff
@@ -41,9 +42,14 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 
-public class ModelGroupStepTests extends OpenSearchTestCase {
+public class RegisterModelGroupStepTests extends OpenSearchTestCase {
```
Comment (owaiskazi19) [inline src/test/java/org/opensearch/flowframework/workflow/RegisterModelGroupStepTests.java:45]: :+1:

**Review by owaiskazi19 — APPROVED** [review]
LGTM

**Review by amitgalitz — APPROVED** [review]

**Pull request merged**